hexrd/core/convolution/convolve.py

```diff
@@ -116,6 +116,7 @@ def _copy_input_if_needed(
     return output
 
 
+# TODO: This function does not work when array or kernel are 32-bit float types.
```

Comment:
This function was copied from `astropy`. It does tend to be one of the more time-consuming functions that run for things like SNIP algorithms. If `astropy` ever makes significant performance improvements to their code, we may want to copy it again from. [Here](https://github.com/astropy/astropy/blob/8407999759909e5e24cc45fa69bfc5c936564d75/astropy/convolution/convolve.py#L124) is the latest version.

We might should avoid from deviating too far from their code, since we may rely on them for updates.